### PR TITLE
Button styles fix

### LINKED ISSUE #256: Icons are expanding loading button height
After some changes in e5a2182, icons used in buttons are throwing off the padding, height, and alignment (in button groups).
screen shot 2016-04-11 at 9 21 25 am

### PR DESCRIPTION
**Pull request by iheartrachie**
Fixes #256 and reworks Button and ButtonGroup components to use the `Object.assign()` pattern on styles. Radium is still required on the Button component so I didn't remove it there. Note: button height goes from 31px to 29px which is more in line with the style guide anyway.

### CODE REVIEW

**Comment by cerinman** [comment]
:shipit:

File: src/components/ButtonGroup.js [hunk 1]
```diff
@@ -90,4 +89,4 @@ const ButtonGroup = React.createClass({
   }
 });
 
-module.exports = Radium(ButtonGroup);
```
Comment (scotttball) [inline src/components/ButtonGroup.js]: We still need Radium for the button `:hover`

Comment (scotttball) [inline src/components/ButtonGroup.js]: and for line 46

File: src/components/Button.js [hunk 2]
```diff
@@ -40,7 +40,7 @@ const Button = React.createClass({
         {this.props.isActive ? (
           <div>
             <Spin direction='counterclockwise'>
-                <Icon size='20' style={[styles.icon, styles.spinner]} type='spinner' />
+                <Icon size='20' style={styles.spinner} type='spinner' />
```
Comment (scotttball) [inline src/components/Button.js]: `<Icon size={20} style={styles.spinner} type='spinner' />`

Comment (iheartrachie) [inline src/components/Button.js]: This should be overwritten when you push your React 15 changes up.

Comment (scotttball) [inline src/components/Button.js]: also, if Luke hasn't changed it already, the indent is off

**Pull request merged**